### ISSUE TITLE
[4.13.1] Unable to share to group from another app when not listed in recent chats.

### ISSUE THREAD

**Issue by bme13** [signalapp/Signal-Android#7202]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
When I try to share from an another app I cannot share to a group when it's not listed in the recent chat history. The recent chats only contains 5 items, which is a mixture of contacts and groups. When I scroll down to the contacts there are no groups. Also searching for the group did not find it.

### Steps to reproduce
- open an other app
- select share with signal
- try to share to a group, not listed in the recent chats.

**Actual result:** Unable to find the group.
**Expected result:** Should be able to select the group. I guess the best solution would be that the group is listed in the contact list. Which seems to be tracked here #6053

### Device info

**Device:** Samsung SM-G900F
**Android version:** 6.0.1
**Signal version:** 4.13.1

**Comment by nrizzio** [comment]
This also applies when forwarding an image, a text, or other message within the app.

**Comment by moxie0** [comment]
Is this a regression?

**Comment by nrizzio** [comment]
@moxie0 I would argue yes. Previously, to share to a group, it would only have to be unarchived. Now, you have to have had recently sent a message to that group.

**Comment by bme13** [comment]
@moxie0, Yes. Prior to 4.13 I didn't have problems with sharing to groups. @nrizzio, your right that the problem is with sharing in general, so not only from external apps. I'm not sure it has anything to do with archiving though (never needed to do that in order to send to a group). I think that in previous versions the share window just showed all the conversations while with the new share look you only see the 5 most recent chats. So when the group is not within the 5 most recent conversations you won't be able to select it. 
One thing I've noticed though is that within the share window the scrollbar seems to be located at a strange place. It's not completely to the right of the screen (see arrow in attached image) while in other screens it's always at the utmost right position. Does the recent chats suppose to have a separate scroll functionality so that you are able to scroll in that upper part and that it's simply not handled correctly on my screen?
![signal2](https://user-images.githubusercontent.com/25846829/33009226-7b6a3eda-cdd6-11e7-881b-e370a425a194.png)

**Comment by GeneKar** [comment]
Furthermore, groups names are not returned when searched for. 

The 2 solution if the group is not in the recent list:
1) bump the thread first with text
2) put the sharable item on the clipboard and then paste

**Comment by Wikinaut** [comment]
Signal "Group(s)" should be sorted-into the alphabetical list of regular Signal contacts.

**Comment by b2pb** [comment]
I also liked the earlier style of sharing better... I can no longer quickly share to groups I want, and only showing the last five contacts means I find myself scrolling through my contact list, or typing more than I used to, to find the contacts I want to share w/.

**Comment by kyleedwardsny** [comment]
I am actively working on a fix for this issue. I should have a PR up some time next week.

**Comment by kyleedwardsny** [comment]
I have a working fix at #7427. Testers welcome.

**Comment by Wikinaut** [comment]
@moxie0 My really very kind question: when can users expect to have this "cannot-share-with-groups"-issue solved?

**Comment by Wikinaut** [comment]
@moxie0 Sorry, this is not fully fixed in the 4.18.0 version: 

When I want to "share" something with Signal, I see the list of
* the last five messages, and 
* the list of the contacts, but
* I still do *not* see the list of groups.

I admit, that I can now search for a certain group and can then find the wanted group in the list of the results. In that way, the 4.18.0 is a small improvement.

**Comment by dcherian** [comment]
Also, MMS groups are not included in search results. Makes for very confusing UX

**Comment by b2pb** [comment]
I see this is closed, but could we please reconsider this new sharing activity. I still feel the earlier way of sharing was better than everything we've seen since ~last October. 

We already have two standards that everyone uses all the time, and that earlier sharing used, the conversation list and the new conversation. ~90% of my messaging happens from the conversation list. I do not know why we've started a new standard, and now a new UX in the share activity. 

In many of my Signal groups we set expiring messages, so my groups inevitably end up @ the bottom of the conversation list. It was simple to tap the share arrow, scroll to the bottom of my conversation list, and share. Now, I hardly ever see my groups as they are hardly ever in 'recent chats,' since they expire. I really don't want to type a group name every time I want to share to existing groups, now a different/new UX from the rest of Signal. Sharing to MMS groups in the conversation list was never an issue earlier.

As I said in an earlier comment, I'm find myself typing or scrolling more than I ever had to in this new share activity. I feel there is too much friction in this new share UI compared to the earlier way of sharing, which showed the conversation list, and if needed, the new conversation.

Sadly, I'm pressing that share arrow less and less, now saving the media to my mobile, then conversation list > conversation, and attaching, if only because that conversation list and new conversation is what we use, all the time.

In saying this, I certainly appreciate the time spent thinking about and testing better ways of sharing in Signal.

**Comment by greyson-signal** [comment]
Hey all, I agree that the sharing UI can be improved. #7577 added the ability to find groups via search to at least make sharing to non-recent groups possible again, but it was probably premature to close this issue entirely. We're going to take all of this feedback into consideration and come up with an updated view for a future release. Thanks!

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.

**Comment by Wikinaut** [comment]
(very kindly asking as I very often need to share with group/s: when will this issue be solved?)

**Comment by pgerber** [comment]
This issue is partly resolved in 4.19. By which I mean, you'll see groups in the search result but they won't be listed otherwise. This is a bit confusing, particularly for people with only a hand full of contacts which can be scrolled through easily. I wouldn't try to search for a group in such a case. The good news is, as mentioned in [a comment above](https://github.com/signalapp/Signal-Android/issues/7202#issuecomment-377987778), this will likely be improved in future versions.

**Comment by Wikinaut** [comment]
@pgerber In my Pull request https://github.com/signalapp/Signal-Android/pull/7416 I made an "ad-hoc" fix (read: mitigation) for the issue in that I simply increased the number of the listed recent chats from 5 to something. This helps a lot, because the probability is higher that a "group" I want to share with is listed.

Perhaps the Signal Team wants to consider to have this value (currently 5) to be increased, or to become a user-option (default 5)?
